V2 : primitives revisted

z3dev opened the pull request:
These changes improve the checking of parameters (options) in the primitives.

NOTE: There are also changes to the tests suites to comply with coding standards.

Fixes #633 
Fixes #626 

### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [X] Does your submission pass tests?

z3dev commented:
There's a lot of redundant code in the primitives, especially the redundant error messages/strings. I can see two ways to address this.

#1 - create a set of common functions to perform common checks, e.g. checkCenter()

#2 - keep the unique checks in the code, and create a set of common functions to throw common errors

Perferences anyone? Suggestions?

SimonClark commented:
> There's a lot of redundant code in the primitives, especially the redundant error messages/strings. I can see two ways to address this.
> 
> 1. create a set of common functions to perform common checks, e.g. checkCenter()
> 2. keep the unique checks in the code, and create a set of common functions to throw common errors
> 
> Perferences anyone? Suggestions?

I think I'd prefer to see a set of options validator functions.  Make for more readable code imo

z3dev commented:
@SimonClark thanks.

I implemented some common checks (actually more complete), and reworked the code for the primitives. Looks better, more readable, but I still don't like the redundant comments.

SimonClark commented on packages/modeling/src/primitives/arc.js:
```diff
@@ -27,13 +29,11 @@ const arc = (options) => {
-
-  if (startAngle < 0 || endAngle < 0) throw new Error('the start and end angles must be positive')
-  if (segments < 4) throw new Error('segments must be four or more')
+  if (!isArray(2, center)) throw new Error('center must be an array of X and Y values')
```
inconsistency in order of parameters here.  Should the thing being checked always be first?

SimonClark commented on packages/modeling/src/primitives/commonChecks.js:
```diff
@@ -0,0 +1,19 @@
+}
+
+// verify that the value is a Number greater than the constant
+const isGT = (value, constant) => (Number.isFinite(value) && value > constant)
```
I have a preference for isGreaterThan over isGT.  Minor thing, but it makes the code more like plain english.  Take it or leave it. :)

z3dev replied:
I actually wrote the changes originally with isGreaterThan() but then I needed isGreaterThanOrEquals() which just seemed moronic for a simple check. It did look up abbreviations for these. :)

SimonClark commented on packages/modeling/src/primitives/commonChecks.js:
```diff
@@ -0,0 +1,19 @@
+// verify that the array has the given dimension, and contains Number values
+const isArray = (dimension, array) => {
```
isArrayOfSize()?
